app/helpers/view_helper.rb

```diff
@@ -6,7 +6,11 @@ def govuk_back_link_to(url = :back, body = 'Back')
 
     url = back_link_url if url == :back
 
-    if url.is_a?(String) && url.end_with?(candidate_interface_application_form_path)
+    if url.to_s.end_with?(candidate_interface_continuous_applications_details_path)
+      body = 'Back to your details'
+    elsif url.to_s.end_with?(candidate_interface_continuous_applications_choices_path)
+      body = 'Back to your applications'
+    elsif url.to_s.end_with?(candidate_interface_application_form_path)
       body = 'Back to application'
     end
 
```

Comment:
I think something like this could also work:

```ruby
body = case URI(url).path
        when candidate_interface_continuous_applications_details_path
          'Back to your details'
        when candidate_interface_continuous_applications_choices_path
          'Back to your applications'
        when candidate_interface_application_form_path
          'Back to application'
        else
          body
        end
```